modules/openapi/src/alloy/openapi/JsonUnknownMapper.scala

```diff
@@ -16,35 +16,66 @@
 package alloy.openapi
 
 import software.amazon.smithy.jsonschema.JsonSchemaMapper
-import software.amazon.smithy.jsonschema.JsonSchemaConfig
 import software.amazon.smithy.jsonschema.Schema.Builder
 import software.amazon.smithy.model.node.Node
-import software.amazon.smithy.model.shapes.Shape
 import alloy.JsonUnknownTrait
 
 import scala.jdk.CollectionConverters._
+import scala.jdk.OptionConverters.*
+import software.amazon.smithy.jsonschema.JsonSchemaMapperContext
+import alloy.DiscriminatedUnionTrait
+import software.amazon.smithy.jsonschema.Schema
 
 class JsonUnknownMapper() extends JsonSchemaMapper {
   private final val ADDITIONAL_PROPERTIES = "additionalProperties"
 
   override def updateSchema(
-      shape: Shape,
-      schemaBuilder: Builder,
-      config: JsonSchemaConfig
+      context: JsonSchemaMapperContext,
+      schemaBuilder: Builder
   ): Builder = {
-    val jsonUnknownMemberName = shape
-      .getAllMembers()
-      .asScala
-      .collect {
-        case (name, member) if member.hasTrait(classOf[JsonUnknownTrait]) =>
-          name
-      }
-      .headOption
+    val shape = context.getShape()
 
-    jsonUnknownMemberName.fold(schemaBuilder) { name =>
+    if (!shape.members.asScala.exists(_.hasTrait(classOf[JsonUnknownTrait])))
       schemaBuilder
-        .removeProperty(name)
-        .putExtension(ADDITIONAL_PROPERTIES, Node.from(true))
+    else {
+      val unknownMember = shape
+        .members()
+        .asScala
+        .find(_.hasTrait(classOf[JsonUnknownTrait]))
+        .getOrElse(
+          sys.error("Didn't find an unknown member, even though we just did")
+        )
+
+      if (shape.isStructureShape()) {
+        schemaBuilder
+          .removeProperty(unknownMember.getMemberName)
+          .additionalProperties(Schema.fromNode(Node.objectNode()))
```

Comment:
OAS 3.0 docs say this is equivalent to a true: https://swagger.io/docs/specification/v3_0/data-models/dictionaries/#free-form-objects

Reply:
actually, gonna roll back to just a true node